python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py

```diff
@@ -0,0 +1,252 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from collections.abc import AsyncGenerator
+from typing import Any
+
+from azure.ai.inference import load_client as load_client_sync
+from azure.ai.inference.aio import ChatCompletionsClient as ChatCompletionsClientAsync
+from azure.ai.inference.models import (
+    AssistantMessage,
+    AsyncStreamingChatCompletions,
+    ChatChoice,
+    ChatCompletions,
+    ChatRequestMessage,
+    ModelInfo,
+    ModelType,
+    StreamingChatChoiceUpdate,
+    SystemMessage,
+    ToolMessage,
+    UserMessage,
+)
+from azure.core.credentials import AzureKeyCredential
+from pydantic import ValidationError
+
+from semantic_kernel.connectors.ai.azure_ai_inference import (
+    AzureAIInferenceChatPromptExecutionSettings,
+    AzureAIInferenceSettings,
+)
+from semantic_kernel.connectors.ai.chat_completion_client_base import ChatCompletionClientBase
+from semantic_kernel.contents.chat_history import ChatHistory
+from semantic_kernel.contents.chat_message_content import ChatMessageContent
+from semantic_kernel.contents.streaming_chat_message_content import StreamingChatMessageContent
+from semantic_kernel.contents.utils.author_role import AuthorRole
+from semantic_kernel.contents.utils.finish_reason import FinishReason
+from semantic_kernel.exceptions.service_exceptions import ServiceInitializationError
+
+MessageConverter: dict[AuthorRole, Any] = {
+    AuthorRole.SYSTEM: SystemMessage,
+    AuthorRole.USER: UserMessage,
+    AuthorRole.ASSISTANT: AssistantMessage,
+    AuthorRole.TOOL: ToolMessage,
+}
+
+
+class AzureAIInferenceChatCompletion(ChatCompletionClientBase):
+    """Azure AI Inference Chat Completion Service."""
+
+    client: ChatCompletionsClientAsync
+
+    def __init__(
+        self,
+        api_key: str | None = None,
+        endpoint: str | None = None,
+        service_id: str | None = None,
+        env_file_path: str | None = None,
+        env_file_encoding: str | None = None,
+        client: ChatCompletionsClientAsync | None = None,
+        model_info: ModelInfo | None = None,
```

Comment:
This is needed since we can get the model info synchronously from an async client. And model info in needed to determine the model name.

Reply:
I'm more thinking about the developer, he now would need to figure out where ModelInfo comes from (if he wants to supply their own client that is a more advanced scenario). In case a self-created client is used, then this is already present, so we shouldn't need it for that case, right? and looking through the fields in ModelInfo, the model_name should be supplied in SK by ai_model_id (common field across ServiceClients) and then model_type and provider can be optional fields. And then see my note on the ordering in init below

Reply:
Ok, I have dropped the model info after confirming with the intended usage of the `load_client` method from the Azure team. Essentially, we don't need to check if the model endpoint provided to the chat client is a chat completion model (same for other agents). The error will surface when the first request is sent, which is a common pattern.